QCA9888: Board file for Yuncore WB5G08

grosjo opened the pull request:
qca9888 : add BDF for Kuwfi WB5G08
Signed-off-by: Joan Moreau <jom@grosjo.net>

grosjo commented:
@DragonBluep @testuser7 @hauke all good ?

grosjo commented:
Ther seems to be something wrong in the " maneconfig"  of this item

![image](https://github.com/user-attachments/assets/ec003728-6868-4e73-9c43-207d31d449fd)

grosjo commented:
and it seems the board-2 file in the openwrt gto is not the correct one

grosjo commented:
@testuser7 @DragonBluep how to fix all this and push the proper data to the main git ?

testuser7 commented:
After merging this PR you need to update the version in the main repository: https://github.com/openwrt/openwrt/blob/main/package/firmware/ipq-wifi/Makefile#L9-L11

testuser7 commented:
Which name is currently used?
- `bus=pci,vendor=168c,device=0056,subsystem-vendor=0000,subsystem-device=0000,variant=KuWfi-WB5G08`
or
- `bus=pci,bmi-chip-id=0,bmi-board-id=31,variant=KuWfi-WB5G08`

grosjo commented:
> After merging this PR you need to update the version in the main repository: https://github.com/openwrt/openwrt/blob/main/package/firmware/ipq-wifi/Makefile#L9-L11

Don't understand what do you mean ? How can I update the main repo I have no access to ?
What "version"  are you speaking about ? With what variables ?

grosjo commented:
> Which name is currently used?
> 
> * `bus=pci,vendor=168c,device=0056,subsystem-vendor=0000,subsystem-device=0000,variant=KuWfi-WB5G08`
>   or
> * `bus=pci,bmi-chip-id=0,bmi-board-id=31,variant=KuWfi-WB5G08`

The onein the board-2.bin file !

bus=pci,vendor=168c,device=0056,subsystem-vendor=0000,subsystem-device=0000,variant=kuwfi_wb5g08

grosjo commented:
> There seems to be something wrong in the "make menuconfig" of this item
> 
> ![image](https://private-user-images.githubusercontent.com/2287111/415911787-ec003728-6868-4e73-9c43-207d31d449fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAyNjg1MzksIm5iZiI6MTc0MDI2ODIzOSwicGF0aCI6Ii8yMjg3MTExLzQxNTkxMTc4Ny1lYzAwMzcyOC02ODY4LTRlNzMtOWM0My0yMDdkMzFkNDQ5ZmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjJUMjM1MDM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzFlMGVkZjY0YjI2NWVmYTVhNTE0OTMwNmE4MmY1MTdlMWZiMzlhNmI5NjU1ZjI1MTczNmZhNmI3YjA5OTNmZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.vbpxXRfhqdBEXNo49zBBOydj-ZnzNlQwyujFsq58pWM)

and how to fix that ?

testuser7 commented:
> How can I update the main repo I have no access to ?

By making changes in your second PR.

> How I can " merge" anything here ?

You can't, I can't either. You must wait.

> What "version" are you speaking about ? With what variables ?

```
PKG_SOURCE_DATE:=2025-02-16 - latest commit date in this repo
PKG_SOURCE_VERSION:=331dd0ead646f0d3fe38cb4bf83cce4b448343e8 - latest commit hash in this repo
PKG_MIRROR_HASH:=f49b3acd56a1cf4a0d52413b060d6adbd65c2754b44cabe096fed69172febbb5 - repository checksum in .tar.zst format
```

> variant=kuwfi_wb5g08

The variant should be set as `KuWfi-WB5G08` in the board file as well as in dts. Why did you change it?

> and how to fix that ?

There is nothing wrong here.

grosjo commented:
Menuconfig : You can see in screenshot that menu is totally upside down. How to fix that ?

Board : I aligned as you requested, now will change again, but we dod that already so many times. Can't we just keep on what's work instead of all this lost time ?

I don't understand your "version" topic. Can you please clarify ?

Bottom-line : How to get this PR pushed in the actual Openwrt tree ?

Thank you

grosjo commented:
I put back the name as "Kuwfi-WB5G08" in the dts and in the board file, still not working....

How to please resolve all these issues (not in my hands !)

grosjo commented:
It looks like whatever changes I make here, it does not go anywhere

Error is now
`Sat Feb 22 13:43:37 2025 kern.err kernel: [   17.938676] ath10k_pci 0000:00:00.0: failed to fetch board data for bus=pci,vendor=168c,device=0056,subsystem-vendor=0000,subsystem-device=0000,variant=KuWfi-WB5G08 from ath10k/QCA9888/hw2.0/board-2.bin
`

How to have things back in order ?

testuser7 commented:
Does it work if you put the `board-2.bin` file manually into the `/lib/firmware/ath10k/QCA9888/hw2.0/` directory?

grosjo commented:
If I pu t my own file manually : yes

If I "git pull" the tree, now
The git on that part seems totally broken (see screenshot for the UI and board data is obviously incorrect)

grosjo commented:
```
$ python3 ../qca-swiss-army-knife/tools/scripts/ath10k/ath10k-bdencoder -i board-kuwfi_wb5g08.qca9888 
FileSize: 12204
FileCRC32: a107039c
FileMD5: 9e9dff02bc1f8ded13260a312453a3cb
BoardNames[0]: 'bus=pci,vendor=168c,device=0056,subsystem-vendor=0000,subsystem-device=0000,variant=KuWfi-WB5G08'
BoardLength[0]: 12064
BoardCRC32[0]: e5508031
BoardMD5[0]: 574b397611c8ffea48731f728cf41f8c
```

testuser7 commented:
Before using the board file this PR needs to be merged. Then you need to update the versions in the `ipq-wifi` package as I mentioned earlier. See for example here: https://github.com/openwrt/openwrt/pull/17463/commits/07f8319d2d9166a41c2fd94ec1422bcd80ada991#diff-190af563813e0feafc3a28a32eeedac9df73be962754d836bf29932a7ed1eab5R9

grosjo commented:
and all those weird numbers come from where ? it does not look device dependant, so really don't understand anything here

Can't we just merge that with the proper board file instead of all this mess, including the  menuconfig UI being totally broken?

Leo-PL commented:
@grosjo the proper way would be to submit the board file to ath10k-devel as per here: https://wireless.docs.kernel.org/en/latest/en/users/drivers/ath10k/boardfiles.html - but maintainers seem to be busy - my submission for ZTE MF286A/R/C waits for a few months now.

grosjo commented:
Hi @testuser7 @DragonBluep 
How to get this moving ??
thanks

grosjo commented:
Hi @testuser7 @DragonBluep Hi @testuser7 @DragonBluep
How to get this moving ??
thanks

grosjo commented:
Maybe @mrnuke ?

mrnuke commented:
@grosjo, I don't have commit rights.

robimarko commented:
Can you post lspci -vv output?

robimarko commented:
> > Can you post lspci -vv output?
> 
> lspci is not available

Obviously you gotta install `pciutils` for it

grosjo commented:
```
# lspci 
00:00.0 Class 0280: 168c:0056


root@OpenWrt:~# dmesg 
[    0.000000] Linux version 6.6.90 (root@gjserver) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 14.2.0 r29667-12c93f2295) 14.2.0, GNU ld (GNU Binutils) 2.42) #0 Mon May 19 08:35:23 2025
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019750 (MIPS 74Kc)
[    0.000000] MIPS: machine is Yuncore WB5G08
[    0.000000] SoC: Qualcomm Atheros QCA956X ver 1 rev 0
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Kernel command line: console=ttyS0,115200n8 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32512
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 121424K/131072K available (5657K kernel code, 570K rwdata, 776K rodata, 1244K init, 190K bss, 9648K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 775.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 4932285024 ns
[    0.000001] sched_clock: 32 bits at 388MHz, resolution 2ns, wraps every 5541893118ns
[    0.008290] Calibrating delay loop... 385.84 BogoMIPS (lpj=1929216)
[    0.074808] pid_max: default: 32768 minimum: 301
[    0.085849] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.093569] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.109205] RCU Tasks Trace: Setting shift to 0 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=1.
[    0.122400] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.132818] futex hash table entries: 256 (order: -1, 3072 bytes, linear)
[    0.140041] pinctrl core: initialized pinctrl subsystem
[    0.147615] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.154410] thermal_sys: Registered thermal governor 'step_wise'
[    0.156143] /ahb/apb: Fixed dependency cycle(s) with /ahb/apb/interrupt-controller@18060010
[    0.182324] clocksource: Switched to clocksource MIPS
[    0.197037] NET: Registered PF_INET protocol family
[    0.202453] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.210824] tcp_listen_portaddr_hash hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.219807] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.227988] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.236094] TCP bind hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.243574] TCP: Hash tables configured (established 1024 bind 1024)
[    0.250412] UDP hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.257372] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.265369] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.271405] PCI: CLS 0 bytes, default 32
[    0.278849] workingset: timestamp_bits=30 max_order=15 bucket_order=0
[    0.286144] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.292300] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.307437] pinctrl-single 1804002c.pinmux: 544 pins, size 68
[    0.314671] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.322449] printk: console [ttyS0] disabled
[    0.327508] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 11, base_baud = 1562500) is a 16550A
[    0.336772] printk: console [ttyS0] enabled
[    0.345843] printk: bootconsole [early0] disabled
[    0.371558] spi-nor spi0.0: w25q128 (16384 Kbytes)
[    0.376653] 4 fixed-partitions partitions found on MTD device spi0.0
[    0.383273] OF: Bad cell count for /ahb/spi@1f000000/flash@0/partitions
[    0.390134] OF: Bad cell count for /ahb/spi@1f000000/flash@0/partitions
[    0.397281] Creating 4 MTD partitions on "spi0.0":
[    0.402247] 0x000000000000-0x000000040000 : "u-boot"
[    0.410157] 0x000000040000-0x000000050000 : "u-boot-env"
[    0.417394] 0x000000050000-0x000000ff0000 : "firmware"
[    0.424584] 2 uimage-fw partitions found on MTD device firmware
[    0.430729] Creating 2 MTD partitions on "firmware":
[    0.435903] 0x000000000000-0x000000220000 : "kernel"
[    0.442781] 0x000000220000-0x000000fa0000 : "rootfs"
[    0.449435] mtd: setting mtd4 (rootfs) as root device
[    0.454788] 1 squashfs-split partitions found on MTD device rootfs
[    0.461185] 0x0000005d0000-0x000000fa0000 : "rootfs_data"
[    0.468427] 0x000000ff0000-0x000001000000 : "art"
[    0.475178] OF: Bad cell count for /ahb/spi@1f000000/flash@0/partitions
[    1.176664] switch0: Atheros AR8337 rev. 2 switch registered on mdio.0
[    1.834125] ag71xx-legacy 19000000.eth: connected to PHY at mdio.0:00 [uid=004dd036, driver=Atheros AR8216/AR8236/AR8316]
[    1.846100] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: sgmii
[    1.852931] i2c_dev: i2c /dev entries driver
[    1.860469] NET: Registered PF_INET6 protocol family
[    1.875388] Segment Routing with IPv6
[    1.879261] In-situ OAM (IOAM) with IPv6
[    1.883491] NET: Registered PF_PACKET protocol family
[    1.888792] 8021q: 802.1Q VLAN Support v1.8
[    1.907130] PCI host bridge to bus 0000:00
[    1.911389] pci_bus 0000:00: root bus resource [mem 0x12000000-0x13ffffff]
[    1.918545] pci_bus 0000:00: root bus resource [io  0x0000]
[    1.924314] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    1.932550] pci 0000:00:00.0: [168c:0056] type 00 class 0x028000
[    1.938796] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x001fffff 64bit]
[    1.945954] pci 0000:00:00.0: PME# supported from D0 D3hot
[    1.952647] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
[    1.959515] pci 0000:00:00.0: BAR 0: assigned [mem 0x12000000-0x121fffff 64bit]
[    1.981815] clk: Disabling unused clocks
[    1.995764] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
[    2.009785] Freeing unused kernel image (initmem) memory: 1244K
[    2.015938] This architecture does not have kernel memory protection.
[    2.022603] Run /sbin/init as init process
[    2.026830]   with arguments:
[    2.026834]     /sbin/init
[    2.026839]   with environment:
[    2.026844]     HOME=/
[    2.026848]     TERM=linux
[    2.848107] init: Console is alive
[    2.852010] init: - watchdog -
[    3.994179] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    4.069274] loop: module loaded
[    4.074636] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.083300] init: - preinit -
[    7.642351] random: crng init done
[    9.221382] Atheros AR8216/AR8236/AR8316 mdio.0:00: Port 2 is up
[    9.229664] eth0: link up (1000Mbps/Full duplex)
[   13.550222] jffs2_scan_eraseblock(): End of filesystem marker found at 0x10000
[   13.557767] jffs2_build_filesystem(): unlocking the mtd device... 
[   13.557787] done.
[   13.566163] jffs2_build_filesystem(): erasing all blocks after the end marker... 
[   55.630525] done.
[   55.640299] jffs2: notice: (484) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[   55.657671] mount_root: overlay filesystem has not been fully initialized yet
[   55.670681] mount_root: switching to jffs2 overlay
[   55.688478] overlayfs: upper fs does not support tmpfile.
[   56.207517] eth0: link down
[   56.231813] procd: - early -
[   56.235233] procd: - watchdog -
[   56.841727] procd: - watchdog -
[   56.845416] procd: - ubus -
[   56.970895] procd: - init -
[   57.477462] kmodloader: loading kernel modules from /etc/modules.d/*
[   58.055184] compat: loading out-of-tree module taints kernel.
[   58.061336] Loading modules backported from Linux version v6.12.6-0-ge9d65b48ce1a
[   58.069108] Backport generated by backports.git v6.1.110-1-35-g410656ef04d2
[   58.226775] ath10k_pci 0000:00:00.0: enabling device (0000 -> 0002)
[   58.233612] ath10k_pci 0000:00:00.0: pci irq legacy oper_irq_mode 1 irq_mode 0 reset_mode 0
[   60.044567] ath10k_pci 0000:00:00.0: qca9888 hw2.0 target 0x01000000 chip_id 0x00000000 sub 0000:0000
[   60.054157] ath10k_pci 0000:00:00.0: kconfig debug 0 debugfs 0 tracing 0 dfs 1 testmode 0
[   60.069222] ath10k_pci 0000:00:00.0: firmware ver 10.4-3.9.0.2-00157 api 5 features no-p2p,mfp,peer-flow-ctrl,allows-mesh-bcast,no-ps,iram-recovery crc32 812c4602
[   60.368917] ath10k_pci 0000:00:00.0: board_file api 2 bmi_id N/A crc32 2b205a98
[   62.298478] ath10k_pci 0000:00:00.0: htt-ver 2.2 wmi-op 6 htt-op 4 cal otp max-sta 512 raw 0 hwcrypto 1
[   62.406058] ath: EEPROM regdomain: 0x0
[   62.406085] ath: EEPROM indicates default country code should be used
[   62.406092] ath: doing EEPROM country->regdmn map search
[   62.406106] ath: country maps to regdmn code: 0x3a
[   62.406113] ath: Country alpha2 being used: US
[   62.406119] ath: Regpair used: 0x3a
[   62.425910] kmodloader: done loading kernel modules from /etc/modules.d/*
[   80.230673] eth0: link up (1000Mbps/Full duplex)
[   80.248340] br-lan: port 1(eth0.1) entered blocking state
[   80.253993] br-lan: port 1(eth0.1) entered disabled state
[   80.259604] eth0.1: entered allmulticast mode
[   80.264152] ag71xx-legacy 19000000.eth eth0: entered allmulticast mode
[   80.271122] eth0.1: entered promiscuous mode
[   80.275589] ag71xx-legacy 19000000.eth eth0: entered promiscuous mode
[   80.328870] br-lan: port 1(eth0.1) entered blocking state
[   80.334517] br-lan: port 1(eth0.1) entered forwarding state

```

grosjo commented:
lspci -vv gives the same output

robimarko commented:
Did you install the busybox version and not pciutils?

grosjo commented:
yes, because the pciutils does not compile (conflicts of all sorts)

robimarko commented:
> yes, because the pciutils does not compile (conflicts of all sorts)

pciutils works just fine i used it like a day or two ago.
Busybox lspci is useless

grosjo commented:
> `lspci -vv` please.
> 
> Also, you need to set the variant in the device DTS as well, otherwise it wont match.

Variant is alreay in the DTS (see related PR) ( qcom,ath10k-calibration-variant = "Yuncore-WB5G08"; )

grosjo commented:
@robimarko 

```
root@OpenWrt:~# lspci -vv
00:00.0 Network controller: Qualcomm Atheros Device 0056
	Device tree node: /sys/firmware/devicetree/base/ahb/pcie@18250000/wifi@0,0
	Control: I/O- Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx-
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 0
	Interrupt: pin A routed to IRQ 12
	Region 0: Memory at 12000000 (64-bit, non-prefetchable) [size=2M]
	Capabilities: [40] Power Management version 3
		Flags: PMEClk- DSI- D1- D2- AuxCurrent=375mA PME(D0+,D1-,D2-,D3hot+,D3cold-)
		Status: D0 NoSoftRst- PME-Enable- DSel=0 DScale=0 PME-
	Capabilities: [50] MSI: Enable- Count=1/8 Maskable+ 64bit-
		Address: 00000000  Data: 0000
		Masking: 00000000  Pending: 00000000
	Capabilities: [70] Express (v2) Endpoint, IntMsgNum 0
		DevCap:	MaxPayload 256 bytes, PhantFunc 0, Latency L0s unlimited, L1 <64us
			ExtTag- AttnBtn- AttnInd- PwrInd- RBE+ FLReset- SlotPowerLimit 0W TEE-IO-
		DevCtl:	CorrErr- NonFatalErr- FatalErr- UnsupReq-
			RlxdOrd+ ExtTag- PhantFunc- AuxPwr- NoSnoop-
			MaxPayload 128 bytes, MaxReadReq 512 bytes
		DevSta:	CorrErr+ NonFatalErr- FatalErr- UnsupReq- AuxPwr- TransPend-
		LnkCap:	Port #0, Speed 2.5GT/s, Width x1, ASPM not supported
			ClockPM+ Surprise- LLActRep- BwNot- ASPMOptComp+
		LnkCtl:	ASPM Disabled; RCB 64 bytes, LnkDisable- CommClk-
			ExtSynch- ClockPM- AutWidDis- BWInt- AutBWInt-
		LnkSta:	Speed 2.5GT/s, Width x1
			TrErr- Train- SlotClk+ DLActive- BWMgmt- ABWMgmt-
		DevCap2: Completion Timeout: Not Supported, TimeoutDis+ NROPrPrP- LTR+
			 10BitTagComp- 10BitTagReq- OBFF Via message, ExtFmt- EETLPPrefix-
			 EmergencyPowerReduction Not Supported, EmergencyPowerReductionInit-
			 FRS- TPHComp- ExtTPHComp-
			 AtomicOpsCap: 32bit- 64bit- 128bitCAS-
		DevCtl2: Completion Timeout: 50us to 50ms, TimeoutDis-
			 AtomicOpsCtl: ReqEn-
			 IDOReq- IDOCompl- LTR- EmergencyPowerReductionReq-
			 10BitTagReq- OBFF Disabled, EETLPPrefixBlk-
		LnkCap2: Supported Link Speeds: 2.5GT/s, Crosslink- Retimer- 2Retimers- DRS-
		LnkCtl2: Target Link Speed: 2.5GT/s, EnterCompliance- SpeedDis-
			 Transmit Margin: Normal Operating Range, EnterModifiedCompliance- ComplianceSOS-
			 Compliance Preset/De-emphasis: -6dB de-emphasis, 0dB preshoot
		LnkSta2: Current De-emphasis Level: -6dB, EqualizationComplete- EqualizationPhase1-
			 EqualizationPhase2- EqualizationPhase3- LinkEqualizationRequest-
			 Retimer- 2Retimers- CrosslinkRes: unsupported
	Capabilities: [100 v2] Advanced Error Reporting
		UESta:	DLP- SDES- TLP- FCP- CmpltTO- CmpltAbrt- UnxCmplt- RxOF- MalfTLP- ECRC- UnsupReq- ACSViol-
		UEMsk:	DLP- SDES- TLP- FCP- CmpltTO- CmpltAbrt- UnxCmplt- RxOF- MalfTLP- ECRC- UnsupReq- ACSViol-
		UESvrt:	DLP+ SDES+ TLP- FCP+ CmpltTO- CmpltAbrt- UnxCmplt- RxOF+ MalfTLP+ ECRC- UnsupReq- ACSViol-
		CESta:	RxErr+ BadTLP- BadDLLP+ Rollover- Timeout- AdvNonFatalErr-
		CEMsk:	RxErr- BadTLP- BadDLLP- Rollover- Timeout- AdvNonFatalErr+
		AERCap:	First Error Pointer: 00, ECRCGenCap- ECRCGenEn- ECRCChkCap- ECRCChkEn-
			MultHdrRecCap- MultHdrRecEn- TLPPfxPres- HdrLogCap-
		HeaderLog: 00000000 00000000 00000000 00000000
	Capabilities: [148 v1] Virtual Channel
		Caps:	LPEVC=0 RefClk=100ns PATEntryBits=1
		Arb:	Fixed- WRR32- WRR64- WRR128-
		Ctrl:	ArbSelect=Fixed
		Status:	InProgress-
		VC0:	Caps:	PATOffset=00 MaxTimeSlots=1 RejSnoopTrans-
			Arb:	Fixed- WRR32- WRR64- WRR128- TWRR128- WRR256-
			Ctrl:	Enable+ ID=0 ArbSelect=Fixed TC/VC=ff
			Status:	NegoPending- InProgress-
	Capabilities: [168 v1] Device Serial Number 00-00-00-00-00-00-00-00
	Capabilities: [178 v1] Latency Tolerance Reporting
		Max snoop latency: 0ns
		Max no snoop latency: 0ns
	Kernel driver in use: ath10k_pci
lspci: Unable to load libkmod resources: error -2

```

testuser7 commented:
@robimarko, what determines whether the driver asks for an `bus=pci,bmi-chip-id=0,bmi-board-id=31,variant=KuWfi-WB5G08` or `bus=pci,vendor=168c,device=0056,subsystem-vendor=0000,subsystem-device=0000,variant=KuWfi-WB5G08` name?
- https://github.com/openwrt/firmware_qca-wireless/pull/56#issuecomment-2628940599

grosjo commented:
> Caldata + PCI bus adress determines what it will request.
> 
> BTW, are you sure that this BDF is being used by ath10k, cause the CRC32 showed by driver of the used BDF and what you posted above differ?

I resubmitted the file to git

grosjo commented:
```
$ python3 ../qca-swiss-army-knife/tools/scripts/ath10k/ath10k-bdencoder -i board-yuncore_wb5g08.qca9888 
FileSize: 12208
FileCRC32: 2b205a98
FileMD5: 0e32dea0b58849754fae1f9cc88ec82b
BoardNames[0]: 'bus=pci,vendor=168c,device=0056,subsystem-vendor=0000,subsystem-device=0000,variant=Yuncore-WB5G08'
BoardLength[0]: 12064
BoardCRC32[0]: e5508031
BoardMD5[0]: 574b397611c8ffea48731f728cf41f8c
```

grosjo commented:
@DragonBluep @testuser7 @hauke @robimarko What would be the next step ?

robimarko commented:
Thanks! Rebased on top of main and merged!
